File: 1210 Sum of Consecutive Prime Numbers/solution.c
#include <stdio.h>
#include <stdlib.h>
#define MAX_NUM 10000
#define GET(n) (mark[(n) >> 5] & (1u << ((n) & 31)))
#define SET(n) (mark[(n) >> 5] |= (1u << ((n) & 31)))

unsigned int mark[(MAX_NUM >> 6) + 1];
short *primes, count[MAX_NUM + 1];
int size = 1;

void eulerSieve() {
    primes = (short*)malloc(sizeof(short));
    primes[0] = 2;
    for(short n = 3; n <= MAX_NUM; n += 2) {
        if(!GET(n >> 1)) {
            primes = (short*)realloc(primes, (size + 1) * sizeof(short));
            primes[size++] = n;
        }
        for(int i = 1, temp; (temp = primes[i] * n) <= MAX_NUM; ++i) {
            SET(temp >> 1);
            if(n % primes[i] == 0)
                break;
        }
    }
}

int main() {
    eulerSieve();
    for(int i = 0; i < size; ++i) {
        for(int j = i, sum = 0; j < size && (sum += primes[j]) <= MAX_NUM; ++j)
            ++count[sum];
    }
    int num;
    while(scanf("%d", &num) && num != 0)
        printf("%hd\n", count[num]);
    free(primes);
	return 0;
}

File: 1210 Sum of Consecutive Prime Numbers/solution.java
import java.io.*;
import java.util.*;

public class Main {
    static short MAX_NUM = 10000;
    static boolean[] isComposite = new boolean[(MAX_NUM >> 1) + 1];
    static ArrayList<Short> primes = new ArrayList<>();
	static short[] count = new short[MAX_NUM + 1];
	public static void main(String[] args) throws IOException {
    	eulerSieve();
	    for(int i = 0; i < primes.size(); ++i) {
            for(int j = i, sum = 0; j < primes.size() && (sum += primes.get(j)) <= MAX_NUM; ++j)
                ++count[sum];
        }
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
    	int num;
    	StringBuilder output = new StringBuilder();
    	while(st.nextToken() == StreamTokenizer.TT_NUMBER && (num = (int)st.nval) != 0)
	        output.append(count[num]).append('\n');
	    System.out.print(output);
	}
	
	static void eulerSieve() {
	    primes.add((short)2);
        for(short n = 3; n <= MAX_NUM; n += 2) {
            if(!isComposite[n >> 1])
                primes.add(n);
            for(int i = 1, temp; i < primes.size() && (temp = primes.get(i) * n) <= MAX_NUM; ++i) {
                isComposite[temp >> 1] = true;
                if(n % primes.get(i) == 0)
                    break;
            }
        }
    }
}
